db/auto_roll_logger.cc

```diff
@@ -32,8 +32,19 @@ Status AutoRollLogger::ResetLogger() {
 }
 
 void AutoRollLogger::RollLogFile() {
-  std::string old_fname = OldInfoLogFileName(
-      dbname_, env_->NowMicros(), db_absolute_path_, db_log_dir_);
+  // This function is called when log is rotating. Two rotations
+  // can happen quickly (NowMicro returns same value). To not overwrite
+  // previous log file we increment by one micro second and try again.
+  uint64_t now = env_->NowMicros();
+  std::string old_fname;
+  for (int i = 0; i < 10; i++) {
```

Comment:
It is confusing that we try several times but still doesn't guarantee no conflict. I would suggest we keep increasing until we find a file name without conflict. 
At least we should add comments about what will happen for this situation.